baybe/parameters/categorical.py

```diff
@@ -22,10 +22,13 @@ class CategoricalParameter(DiscreteParameter):
     # See base class.
 
     # object variables
-    # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1197
-    _values: tuple = field(
+    _values: Tuple[str, ...] = field(
         converter=tuple,
-        validator=[min_len(2), validate_unique_values],  # type: ignore
+        validator=(
+            min_len(2),
```

Comment:
Technically, although a bit weird, there could be a parameter that only has a single value. Do we want to allow that?

Reply:
Same question for the other classes, e.g.the custom one.

Reply:
I've asked myself that question already a long time ago (when the parameters file was created) and I think I have a clear answer to it: Yes, having single-valued parameters should be supported, but only via a separate Parameter type, which we could call for instance `FixedParameter`, `StaticParameter`, or `ContextParameter`. The rationale: we want to give users the possibility to explicitly declare the existence of additional single-valued columns in their data (e.g. meta-information of their experiments) and BayBE should swallow those without error. However, these parameter need to be handled differently internally (similar to the TaskParameter), because they would not contribute to the computational representations – their value is fixed, so no need to pass them to any solver.

There is probably already a workitem on Azure, but I can open a PR where we can start working on it. In that same PR, we need to urgently fix a related issue: additional columns are currently simply ignored. We should forbid this and explicitly require either a corresponding regular parameter or "fixed parameter" as described above.

Reply:
TL;DR: no changes required for this PR, but the other PR should follow soon!

Reply:
im not sure we should equate metadata with parameters, we have metadata already, seems more straightwordard to me to make this interface proper and use it for metadata rather than inventing new parameters, in a sense this is exactly what its for and you dont have to make filterings for the comp_rep creation

why would you like to disallow additional columns ? This might mean one more unneccessary step for users

Reply:
Your explanation makes sense to me, so no reason to change anything here. I think it would be good if you could copy your explanation here to a discussion as this is probably the correct place to discuss the questions and concerns regarding this.

Reply:
Don't get me wrong, @Scienfitz, I mean metadata in a different sense. Let answer your question first and then it will become clear:

The reason is it because it can give you the absolute worst silent bugs in terms of DOE, and I accidentally stepped into it already twice in my projects and only noticed during debugging! Assume you simply **forget** to declare a parameter. Then, when reading in measurement data that really depends on those parameter's values, you completely neglect the dependence and get a completely wrong model. Chances that this happens when setting up a new project are medium because that's when you think about your data and design space (still, happened to me!) but significantly higher when you adjust your data collection pipeline later on and suddenly have additional relevant columns (happened to me, too!). I think there is absolutely no downside of requiring explicit definition of all columns because the only thing that can happen is that you raise awareness to the user that there is additional stuff floating around that is unspecified. Adding an explicit and simple column selection line of code is really much better than the alternative. Do you agree?

So back to the "metadata" Parameters. Their main purpose would really be a sanity check. If people know that their experiments are all conducted on "machine A" and this information is contained in the data, then this can help avoiding accidentally passing data from "machine B". You know what I mean?

Reply:
imo this should be moved to github discussions and thread can be closed
as is, theres no parameter that should be allowed a singular value

Reply:
Done, see [here](https://github.com/emdgroup/baybe/discussions/62). Will resolve